src/cli/cache.rs

```diff
@@ -32,12 +32,14 @@ impl<K: Hash + Eq + Send + Sync, V: Clone + Send + Sync> Cache for NativeChronoC
   #[allow(clippy::too_many_arguments)]
   async fn set<'a>(&'a self, key: K, value: V, ttl: NonZeroU64) -> Result<V> {
     let ttl = Duration::from_millis(ttl.get());
-    self
-      .data
-      .write()
-      .unwrap()
-      .insert(key, value, ttl)
-      .ok_or(anyhow!("unable to insert value"))
+    Ok(
+      self
+        .data
+        .write()
+        .unwrap()
+        .insert(key, value.clone(), ttl)
```

Comment:
well, it's an additional cloning for value (that is already happens inside `into_schema.rs`) mostly just to satisfy type?

I wonder why we even return `V` in this method to drop it immediately inside `request_context`? Maybe it'll be better to change interface and drop `V`?

Reply:
maybe I can convert it from Result to Option type. since `TtlCache` returns an Option type so returning it wouldn't have any overhead. I just think it should atleast return the optional type because there might be some use case in the future where that value is required